server/src/services/asset.service.ts

```diff
@@ -372,8 +372,10 @@ export class AssetService {
       return JobStatus.FAILED;
     }
 
-    // Ignore requests that are not from external library job but is for an external asset
-    if (!fromExternal && (!asset.library || asset.library.type === LibraryType.EXTERNAL)) {
+    // Ignore requests that are not from external library job but is for an external read-only asset
+    const isReadOnlyLibrary = !asset.library || asset.library.isReadOnly ||
+      (asset.library.isReadOnly === null && asset.library.type === LibraryType.EXTERNAL);
+    if (!fromExternal && isReadOnlyLibrary) {
```

Comment:
This should just check if from external and the asset is read only and skip if so. No reason to make it complicated.